ldclient/__init__.py

```diff
@@ -12,38 +12,87 @@
 __BUILTINS__ = ["key", "ip", "country", "email",
                 "firstName", "lastName", "avatar", "name", "anonymous"]
 
-
 """Settings."""
-client = None
-sdk_key = None
 start_wait = 5
-config = Config()
 
-_lock = ReadWriteLock()
+__client = None
+__config = Config()
+__lock = ReadWriteLock()
 
 
+# sets config.
+def set_config(config):
+    global __config
+    global __client
+    global __lock
+    try:
+        __lock.lock()
+        if __client:
+            log.info("Reinitializing LaunchDarkly Client " + version.VERSION + " with new config")
+            new_client = LDClient(config, start_wait)
+            old_client = __client
+            __client = new_client
+            old_client.close()
+    finally:
+        __config = config
+        __lock.unlock()
+
+
+# 2 use cases:
+# 1. Initial setup: sets the sdk key for the uninitialized client
+# 2. Allows on-the-fly changing of the sdk key. When this function is called after the client has been initialized
+#    the client will get re-initialized with the new sdk key. In order for this to work, the return value of
```

Comment:
I don't think we should say that it should *never* be assigned; that seems a bit extreme. It is fine to get the client at the beginning of a request handler that needs to get a bunch of flags. You just don't want to keep the reference long-term.  

On the other hand, we believe the population of users who need to change the SDK key at runtime is rather small. So most people can disregard this warning altogether if they like.  I think we can re-word this to be less scary, while still explaining enough of the implications of different use cases.